website/static/js/share/stats.js

```diff
@@ -79,20 +87,107 @@ function timeGraph (data) {
             show: false
         },
         tooltip: {
-            grouped: false,
-            format: {
-              name: function (name, ratio, id, index) {
-                  if (name === 'pubmed') {
-                      name = 'pubmed central';
-                  }
-                  return name; 
-              }
-            }
+            grouped: false
         }
     });
 }
 
-Stats.view = function(ctrl) {
+/* Creates an Elasticsearch aggregation by source */
+Stats.sourcesAgg = function () {
+    var sourcesQuery = {'match_all': {} };
+    var sourcesAgg = {'sources': utils.termsFilter('field', '_type')};
+    return {'query' : sourcesQuery, 'aggregations': sourcesAgg, 'filters' : {}};
+};
+
+/* Creates an Elasticsearch aggregation that breaks down sources by date (and number of things published on those dates) */
+Stats.sourcesByDatesAgg = function () {
+    var dateTemp = new Date(); //get current time
+    dateTemp.setMonth(dateTemp.getMonth() - 3);
+    var threeMonthsAgo = dateTemp.getTime();
+    var dateHistogramAgg = {
+        sourcesByTimes: utils.termsFilter('field', '_type')
+    };
+    dateHistogramAgg.sourcesByTimes.aggregations = {
+        articlesOverTime : {
+            filter: utils.rangeFilter('providerUpdatedDateTime', threeMonthsAgo),
+            aggregations: {
+                articlesOverTime: utils.dateHistogramFilter('providerUpdatedDateTime', threeMonthsAgo)
+            }
+        }
+    };
+    return {aggregations: dateHistogramAgg};
+};
+
+/* Helper function for dealing with epoch times returned by elasticsearch */
+Stats.timeSinceEpochInMsToMMYY = function (timeSinceEpochInMs) {
+    var d = new Date(0);
+    d.setUTCSeconds(timeSinceEpochInMs / 1000);
+    return d.getMonth().toString() + '/' + d.getFullYear().toString().substring(2);
+};
+
+/* Parses elasticsearch data so that it can be fed into a c3 donut graph */
+Stats.shareDonutGraphParser = function (data) {
+    var chartData = {};
+    chartData.name = 'shareDonutGraph';
+    chartData.columns = [];
+    chartData.colors = {};
+    chartData.type = 'donut';
+
+    var providerCount = 0;
+    var hexColors = utils.generateColors(data.aggregations.sources.buckets.length);
+    var i = 0;
+    data.aggregations.sources.buckets.forEach(
+        function (bucket) {
+            chartData.columns.push([bucket.key, bucket.doc_count]);
+            providerCount = providerCount + (bucket.doc_count ? 1 : 0);
+            chartData.colors[bucket.key] = hexColors[i];
+            i = i + 1;
+        }
+    );
+    chartData.title = providerCount.toString() + ' Provider' + (providerCount !== 1 ? 's' : '');
+    $('.c3-chart-arcs-title').text(chartData.title); //dynamically update chart title
```

Comment:
Not that can be done easily. For some reason c3 does not like dynamic titles. The profile charts + mithril stuff that i have been working on gets around this issue... So i propose leaving it here for now until i back port what ive done on the profile page to share?